Cloudinary/Classes/Core/Network/CLDNetworkCoordinator.swift

```diff
@@ -68,6 +69,7 @@ internal class CLDNetworkCoordinator: NSObject {
         let requestParams = params.signed ? getSignedRequestParams(params) : params.params
         var headers :[String : String] = getHeaders()
         headers.cldMerge(extraHeaders)
+        headers.cldMerge(self.extraHeaders) //User's configed extra headers
```

Comment:
It should be the opposite, `headers` should override `self.extraHeaders`

Reply:
and please add tests for these scenarios. 

When you set headers on `CLDCloudinary`, check that they are actually set, then you can override.